news/2016-10-26-tip3-performance.asciidoc

```diff
@@ -0,0 +1,184 @@
+However, if you do `<T> void observe(@Observes ProcessAnnotatedType<T> event)` your extension will be notified for every class that is discovered.
+Now if we have an application with one EXPLICIT bean archive with 2000 classes, then a specific observer method on such extesion will be called 1000x.
+And if there three similar extensions, the CDI container will have to create 2000 events and notify 6000 observers.
+This should be ok unless the observer logic is overly complex.
+
+*CONCLUSION*: Pay attention to extensions used in your application.
+If you wonder what extensions are in service, use for example Weld Probe link:http://probe-weld.itos.redhat.com/weld-numberguess/weld-probe#/observers?bda=undefined&kind=EXTENSION[observers view, window="_blank"] or link:http://probe-weld.itos.redhat.com/weld-numberguess/weld-probe#/events[fired events view, window="_blank"] together with link:http://docs.jboss.org/weld/reference/latest/en-US/html/configure.html#config-dev-mode[`org.jboss.weld.probe.eventMonitor.containerLifecycleEvents` enabled, window="_blank"].
+
+TIP: If possible, don't use the extensions which observe all the annotated types from your application, i.e. which defined something like `void observe(@Observes ProcessAnnotatedType<?> event)`.
+Since CDI 1.1 `@WithAnnotations` should be used to restrict the set of types an extension is willing to process. link:http://probe-weld.itos.redhat.com/weld-numberguess/weld-probe#/observers?bda=undefined&kind=EXTENSION&observedType=javax.enterprise.inject.spi.ProcessAnnotatedType[Weld Probe, window="_blank"] warns you about these extensions.
+
+
+[[jandex]]
+=== Jandex
+
+In some environments (WildFly, Weld SE, and more) Weld can leverage an "offline reflection library", such as link:https://github.com/wildfly/jandex[Jandex, window="_blank"], to speed up the scanning process.
+These libraries allow to effectively filter classes link:http://docs.jboss.org/cdi/spec/1.2/cdi-spec.html#what_classes_are_beans[which are not beans, window="_blank"] and vetoed classes.
+It works like this: Jandex generates an index (scans the bytecode) and so Weld does not even have to load classes and use reflection API to detect types which should be ignored.
+
+TIP: In Weld SE and Servlet it’s also possible to generate the Jandex index beforehand, e.g. using the link:https://github.com/wildfly/jandex#user-content-using-the-ant-task-to-index-your-project[Ant task, window="_blank"].
+
+== Runtime
+
+In runtime the overhead of a framework should be minimal so that it's not an application bottleneck.
+However, CDI needs to do all that magic behind the scenes - create bean instances, manage contexts, intercept and decorate invocations, etc.
+The following sections describe what could be done so that the impact is minimized.
```

Comment:
`could be done so that the impact is minimized.` -> `could be done to minimize the impact.`
Nothing wrong with your version, the one I suggest is just a well-known phrase.